src/FirmwarePlugin/FirmwarePlugin.cc

```diff
@@ -567,3 +567,32 @@ Autotune* FirmwarePlugin::createAutotune(Vehicle *vehicle)
 {
     return new Autotune(vehicle);
 }
+
+void FirmwarePlugin::updateAvailableFlightModes(FlightModeList modeList)
+{
+    _availableFlightModeList.clear();
+
+    for(auto mode: modeList){
+        _updateModeMappings(mode);
+    }
+}
+
+void FirmwarePlugin::_setModeEnumToModeStringMapping(FlightModeCustomModeMap enumToString)
+{
+    _modeEnumToString = enumToString;
+}
+
+void FirmwarePlugin::_updateModeMappings(FirmwareFlightMode &mode){
```

Comment:
I would make this method take in a modeList not a single mode. The reason is that this method expects the list to be empty when it is called. Both those clear calls are in multiple places in other parts of the codebase outside of this method. It would be super easy to forget to call clear before calling this method. By making the method take a list you can clear the list as needed directly in the _updateModeMappings method and outside callers don't need to know to do that themselves.

Reply:
Sure will update this